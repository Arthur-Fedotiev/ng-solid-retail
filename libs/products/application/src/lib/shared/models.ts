import { Categories, Colors, Retailers, Tiers } from './constants';

export type CategoriesMap = typeof Categories;
export type Category = typeof Categories[keyof typeof Categories];

export type RetailersMap = typeof Retailers;
export type TiersMap = typeof Tiers;
export type ColorsMap = typeof Colors;
