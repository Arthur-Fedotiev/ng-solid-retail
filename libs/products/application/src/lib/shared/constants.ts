import {
  ProductCategory,
  ProductRetailer,
  ProductTier,
  SHOES_AND_CLOTHING_COLORS,
} from '@sr/products/entities';

export const Categories = ProductCategory;
export const Retailers = ProductRetailer;
export const Tiers = ProductTier;
export const Colors = {
  [Categories.Shoes]: SHOES_AND_CLOTHING_COLORS,
  [Categories.Clothing]: SHOES_AND_CLOTHING_COLORS,
} as const;
