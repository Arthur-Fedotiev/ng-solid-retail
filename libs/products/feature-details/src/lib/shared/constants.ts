import { Tiers } from '@sr/products/application';

export const PRODUCT_TIERS = Tiers;
