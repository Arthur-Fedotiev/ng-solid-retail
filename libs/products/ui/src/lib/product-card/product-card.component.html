<mat-card class="card mat-elevation-z3" tabindex="0">
  <img
    mat-card-image
    src="{{ product.url }}"
    alt="Photo of {{ product.name }}"
  />
  <p class="card__price-container">
    <span class="price">${{ product.price }}</span>
  </p>
  <mat-card-header class="card__header">
    <mat-card-title [matTooltip]="product.name" class="card__title">{{
      product.name
    }}</mat-card-title>
  </mat-card-header>
  <mat-card-content
    class="card__content"
    fxLayout="row nowrap"
    fxLayoutGap="2rem"
    fxLayoutAlign="start center"
  >
    <div>
      <p>Retailer</p>
      <p>GTIN</p>
    </div>
    <div>
      <p>{{ product.retailer }}</p>
      <p>{{ product.sku }}</p>
    </div>
  </mat-card-content>
</mat-card>
