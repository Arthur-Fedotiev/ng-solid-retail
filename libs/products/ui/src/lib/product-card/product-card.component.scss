.card {
  margin-bottom: 2rem;

  &__header {
    border-bottom: 2px solid lightgrey;
    margin-bottom: 1rem;
  }

  &__price-container {
    padding: 0.25rem;
    margin-left: auto;
    text-align: right;

    .price {
      display: inline-block;
      padding: 0.25rem 0.75rem;
      border-radius: 0.5rem;
      background-color: rgb(75, 121, 247);
      color: #fff;
    }
  }

  &__title {
    font-size: 1rem;
    text-align: center;
  }
}
