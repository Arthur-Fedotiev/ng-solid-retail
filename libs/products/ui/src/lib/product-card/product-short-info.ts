export interface ProductCard {
  product: ProductShortInfo;
}

export interface ProductShortInfo {
  id: string;
  price: number;
  name: string;
  sku: string;
  retailer: string;
  currencySymbol?: string;
}
