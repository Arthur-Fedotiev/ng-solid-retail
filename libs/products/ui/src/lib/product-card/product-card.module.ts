import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductCardComponent } from './product-card.component';
import { SharedUiMaterialModule } from '@omnia/shared/ui-material';
import { FlexLayoutModule } from '@angular/flex-layout';

@NgModule({
  declarations: [ProductCardComponent],
  imports: [CommonModule, SharedUiMaterialModule, FlexLayoutModule],
  exports: [ProductCardComponent],
})
export class ProductCardModule {}
