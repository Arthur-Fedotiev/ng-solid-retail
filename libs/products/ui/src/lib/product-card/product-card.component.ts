import {
  ChangeDetectionStrategy,
  Component,
  Input,
  OnInit,
} from '@angular/core';
import { ProductShortInfo } from './product-short-info';

@Component({
  selector: 'omnia-product-card',
  templateUrl: './product-card.component.html',
  styleUrls: ['./product-card.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductCardComponent implements OnInit {
  @Input() product!: ProductShortInfo;

  constructor() {}

  ngOnInit(): void {}
}
