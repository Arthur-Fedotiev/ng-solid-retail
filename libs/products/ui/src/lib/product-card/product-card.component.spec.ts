import { CommonModule } from '@angular/common';
import {
  ComponentFixture,
  ComponentFixtureAutoDetect,
  TestBed,
} from '@angular/core/testing';
import { SharedUiMaterialModule } from '@omnia/shared/ui-material';

import { ProductCardComponent } from './product-card.component';

describe('ProductCardComponent', () => {
  const productShortInfoStub = {
    id: '1',
    price: 1,
    name: 'Product 1',
    sku: 'SKU 1',
    retailer: 'Retailer 1',
  };

  let component: ProductCardComponent;
  let fixture: ComponentFixture<ProductCardComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ProductCardComponent],
      imports: [SharedUiMaterialModule, CommonModule],
      providers: [{ provide: ComponentFixtureAutoDetect, useValue: true }],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ProductCardComponent);
    component = fixture.componentInstance;

    component.product = productShortInfoStub;

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should receive the product from the parent component', () => {
    expect(component.product).toEqual(productShortInfoStub);
  });

  it('should render the product ', () => {
    expect(fixture).toMatchSnapshot();
  });
});
