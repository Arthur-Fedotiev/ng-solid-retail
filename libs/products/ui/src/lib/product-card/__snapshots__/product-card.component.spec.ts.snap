// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardComponent should render the product  1`] = `
<omnia-product-card
  __ngContext__={[Function Number]}
  product={[Function Object]}
>
  <mat-card
    class="mat-card mat-focus-indicator card mat-elevation-z3"
    tabindex="0"
  >
    <img
      alt="Photo of Product 1"
      class="mat-card-image"
      mat-card-image=""
      src="https://m.media-amazon.com/images/I/711eKF9vJuL._AC_UY695_.jpg"
    />
    <p
      class="card__price-container"
    >
      <span
        class="price"
      >
        $1
      </span>
    </p>
    <mat-card-header
      class="mat-card-header card__header"
    >
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title card__title"
        >
          Product 1
        </mat-card-title>
      </div>
    </mat-card-header>
    <mat-card-content
      class="mat-card-content card__content"
      fxlayout="row nowrap"
      fxlayoutalign="start center"
      fxlayoutgap="2rem"
    >
      <div>
        <p>
          Retailer
        </p>
        <p>
          GTIN
        </p>
      </div>
      <div>
        <p>
          Retailer 1
        </p>
        <p>
          SKU 1
        </p>
      </div>
    </mat-card-content>
    <mat-card-actions
      class="mat-card-actions"
    >
      <button
        class="mat-focus-indicator mat-button mat-button-base"
        mat-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          EDIT
        </span>
        <span
          class="mat-ripple mat-button-ripple"
          matripple=""
        />
        <span
          class="mat-button-focus-overlay"
        />
      </button>
      <button
        class="mat-focus-indicator mat-button mat-button-base"
        mat-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          DELETE
        </span>
        <span
          class="mat-ripple mat-button-ripple"
          matripple=""
        />
        <span
          class="mat-button-focus-overlay"
        />
      </button>
    </mat-card-actions>
  </mat-card>
</omnia-product-card>
`;
